src/validator/test/QueryValidatorTest.cpp

```diff
@@ -1101,6 +1101,19 @@ TEST_F(QueryValidatorTest, GoInvalid) {
         auto result = checkResult(query);
         EXPECT_EQ(std::string(result.message()), "SemanticError: Duplicate Column Name : `id'");
     }
+    {
+        std::string query = "GO FROM \"1\" OVER like, serve YIELD like._dst AS id, serve._src AS "
+                            "id, serve._dst AS DST | GO FROM $-.DST OVER like";
+        auto result = checkResult(query);
+        EXPECT_EQ(std::string(result.message()), "SemanticError: Duplicate Column Name : `id'");
+    }
+    {
+        std::string query =
+            "$a = GO FROM \"1\" OVER * YIELD like._dst AS id, like._src AS id, serve._dst as DST; "
```

Comment:
IMO, According to the discussion if the duplicate alias is not used won't report error.

Reply:
emm, report error even if the duplicate alias is not used

Reply:
Ok, if so need to add case about that two variable one with duplicate alias the other without and the latter sentence use the variable without duplicate only.